test/test_utils_config_module.py

```diff
@@ -395,6 +395,18 @@ def test_reference_is_default(self):
         t["a"] = "b"
         self.assertFalse(config._is_default("e_dict"))
 
+    def test_invalid_config_int(self):
+        with self.assertRaises(AssertionError):
+            _ConfigEntry(
+                Config(default=2, env_name_default="FAKE_DISABLE", value_type=int)
```

Comment:
Can we also test this being invalid via the case when value_type is not set explicitly?

Reply:
When value_type isn't explicitly set, doesn't it default to None, which bypasses the assert?

Reply:
Ah - I for some reason thought that value_type was set in this constructor - it's not, so we should always have a ValueType - https://github.com/pytorch/pytorch/blob/d2a2b9f8a74218d685373e3b942e591dcc8c5c0a/torch/utils/_config_module.py#L210